server/services/database/profiles.js

```diff
@@ -33,7 +33,14 @@ const deleteProfile = (profileId) => {
 
 const getProfileById = (id) => {
 	return pool
-		.query("SELECT * FROM profiles where id = $1", [id])
+		.query("select profiles.*, array_agg(' ' || groups.group_name ) as groups \
+				from profile_group \
+				inner join groups \
+				   on groups.id=profile_group.group_id\
+				full outer join profiles \
+				   on profiles.id=profile_group.profile_id \
+				group by(profiles.id) having profiles.id=$1", 
```

Comment:
The `array_agg` function will create an array of group names from all the rows with the same `profile.id` (because of the `group by profiles.id` so you don't need to use `' ' || groups.group_name` to separate them.

Also this will return an array containing a single `null` for profiles who have no groups. I think the best option is to use `coalesce` as before, to replace :

```
select profiles.*, coalesce(array_agg(group_name) filter (where profile_group.profile_id is not null), '{}')
from profiles
left join profile_group on profile_group.profile_id = profiles.id
left join groups on profile_group.group_id = groups.id
where profiles.id = $1
group by (profiles.id);
```